### PR TITLE
Reimplement prefix trie for Boxer

### LINKED ISSUE #135: [BUG] Inconsistent sorting of the elements in lookup tree
### Description

Because of inconsistent sorting, the lookup trie may return NOT_FOUND even if the action/resource is added.

### Steps to reproduce the issue

Can re reproduced if the path contains something like:
`/jobs/requests/{job_id}`
`/jobs/requests/by-tag/{request_id}`

### Describe the results you expected

The lookup trie always returns the consistent result

### System information

_No response_

### PR DESCRIPTION
**Pull request by s-vitaliy**
Resolves #135

## Scope

Implemented:
- The `trie_rs` crate was replaced with priority-based prefix tree.

## Checklist

- [ ] GitHub issue exists for this change.
- [ ] Unit tests added and they pass.
- [ ] Line Coverage is at least 80%.
- [ ] Review requested on `latest` commit.

### CODE REVIEW

File: src/services/prefix_tree/tests.rs [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+            RequestSegment::Hostname("www.example.com".to_string()),
+            RequestSegment::Verb(HTTPMethod::Get),
+            RequestSegment::Path(PathSegment::Static("api".to_string())),
+            RequestSegment::Path(PathSegment::Static("v1".to_string())),
+            RequestSegment::Path(PathSegment::Static("resources".to_string())),
+            RequestSegment::Path(PathSegment::Parameter),
+            RequestSegment::Path(PathSegment::Static("ids".to_string())),
+            RequestSegment::Path(PathSegment::Parameter),
+        ],
+        // Note: this value is not actually correct, but it still must be handled by the trie
```
Comment (george-zubrienko) [inline src/services/prefix_tree/tests.rs]: correct in what sense?

File: src/services/prefix_tree/tests.rs [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    assert_eq!(value, "value3");
+}
+
+fn wrapped_pretty_assert(expected: String) -> impl Fn(String) {
+    move |actual: String| assert_eq!(actual, expected)
+}
+
+#[test_case("www.example.com/api/v1/resources/resource/resource1" => using wrapped_pretty_assert("value0".to_string()); "with parameter match")]
+#[test_case("www.example.com/api/v1/resources/my-resource" => using wrapped_pretty_assert("value1".to_string()); "with exact match")]
+//  TODO: Not implemented since sample with my-id below overwrites the sample with parameter
```
Comment (george-zubrienko) [inline src/services/prefix_tree/tests.rs]: not sure I understand this TODO

**Review by george-zubrienko — APPROVED** [review]
Two small questions, I don't have any suggestions for more extended tests, but here the AI might actually help to generate more test cases. Esp ask it for edge cases.

File: src/services/prefix_tree/tests.rs (hunk 2 above)
Comment (s-vitaliy) [reply src/services/prefix_tree/tests.rs]: I will fix it before publishing this PR

File: src/services/prefix_tree/tests.rs (hunk 1 above)
Comment (s-vitaliy) [reply src/services/prefix_tree/tests.rs]: We only support parameters in routes, not hostnames
